compiler/rustc_middle/src/ty/relate.rs

```diff
@@ -616,24 +616,28 @@ impl<'tcx> Relate<'tcx> for &'tcx ty::List<ty::ExistentialPredicate<'tcx>> {
         // in `a`.
         let mut a_v: Vec<_> = a.into_iter().collect();
         let mut b_v: Vec<_> = b.into_iter().collect();
-        a_v.sort_by(|a, b| a.stable_cmp(tcx, b));
+        a_v.sort_by(|a, b| a.skip_binder().stable_cmp(tcx, &b.skip_binder()));
```

Comment:
This seems fine, but I think it works only because we are moving towards a "normalized" form where bound things are referenced purely by index, and this binder has exactly one bound item.

Reply:
Might be worth adding a comment about this.

Reply:
(In particular, if we had something like `for<'a>...` and `for<'b>...`, and the late-bound regions referenced them by name, those are equivalent, but they might be sorted differently.)